clients/da-vinci-client/src/main/java/com/linkedin/davinci/store/AbstractStorageEngine.java

```diff
@@ -389,66 +392,137 @@ public synchronized void endBatchWrite(StoragePartitionConfig storagePartitionCo
     }
   }
 
+  private void executeWithSafeGuard(int partitionId, Runnable runnable) {
```

Comment:
This made me think it recursively calling itself... 😅

Reply:
😆